main.py

```diff
@@ -1116,3 +1116,22 @@ async def on_table_enabled(self, event):
         content = {"group_table": self.table_group}
         name = "kytos/mef_eline.enable_table"
         await aemit_event(self.controller, name, content)
+
+    @listen_to('kytos/topology.switch.deleted')
+    def on_switch_deleted(self, event):
+        """Remove EVCs which were affected by the deleted switch"""
+        self.handle_switch_deleted(event)
+
+    def handle_switch_deleted(self, event):
```

Comment:
In terms of code, it's reasonable. But I'm not sure if at the application level, in this next release `2023.2`, given a switch being deleted if EVCs (a service) should be removed and auto archived without the user doing that first. Meaning, network operators might get surprised with this operation, so until we also have the previewing features https://github.com/kytos-ng/topology/issues/149 it would be difficult for users to understand service impact of a deletion. 

Also, it might be worth checking if an intermediary switch that just got deleted if it ended up with failover flows still installed. This issue deserves to be discussed and refined more before being implemented, unless you've already done it and it hasn't been documented yet, let me know if you did.